Trader 2

Kurfursten opened the pull request:
I've actually had this mostly done for a while but I finished it now because I'm having laptop troubles and worried I could lose the code if I don't push it. I do so reluctantly because I think any trader PR is a battle to discuss due to certain metaclubs and I wanted to wait until the end of my FFF goal, but here we are. This is a substantial change to how traders operate.

**Wallets**

Standardized starting money to 150-250 credits and removed free coin. (Important, see alt title info). Previously wallets could have any of these values exactly and no other: 1, 2, 10, 11, 20, 100, 101, 110, 200, 1000, 1001, 1010, 1100, 2000. Total extremes. Now they can have any amount of money in the range of 150 to 250. On average this is probably a slight nerf but it's consistency.

**Alt Titles/Antagonism**

* Traders of any kind will never be starting revheads or revsquad heads. This seems obvious as they are not crew.
* Vagabond and Traveler alt titles removed. They seem to be the go-to shitbird titles and I think in general they encourage an image of raiding for loot.
* Trader is closest to legacy trader. Starts with the thermos, box of donk pockets, and trader coin.
* Merchant gives up the opportunity to start with a coin (important, see vendor changes) but instead gets a loyalty implant and can **never** start as a traitor, double agent, vampire, changeling, or cultist. The reasoning being to get the specially approved merchant status, you need to pass a difficult background check. Additionally, a fax with their information will be sent to the station immediately when they join. The idea is to give up some comforts in exchange for credibility. Note they could *become* an antagonist through autotator, thralling, deimplanting+rev/cult, etc.
![untitled](https://user-images.githubusercontent.com/9782036/45190723-1283ac80-b205-11e8-9bf0-ff9dfeb36746.png)



**Vendor Changes**
* Items are no longer removed from the vendor at roundstart by RNG
* Increased random borg modules from 1 to 3 (no repeats)
* Other than borg modules, everything costs money - hence, the only thing to use coins for are borg items
* The machine is unhackable (but not unemaggable) meaning you can not use it without a trader ID and you can not abuse the throw wire.
* I'm open to adjusting costs but I put them all over the map, and more trader items are on the way soon(TM). Some like the AMS and WW are rightfully expensive whereas niche toys like the Sea Devil are cheap.

_The hope in all of this is to give traders more incentive to trade and to behave. They can choose to signal they are 'NT supportive' by being Merchants or remain shady dealers as Traders. Similarly, now they have reason to acquire more money: to buy more products to sell. Tradeoffs include some lost start items, no longer a chance to start with 1k or even 2k credits, and ideally a curbing of bad behaviors through title reworks._

🆑 
* tweak: Traders have been overhauled. Merchants are now more officially recognized, but Traders get some small perks for remaining free agents. Vagabonds and travelers have been shooed off.
* tweak: Trader items now cost credits and are not randomized.
* tweak: The trader vendor can no longer be hacked.
* tweak: Merchants now spawn with NT loyalty implants and are barred from many antagonist roles. IAAs should keep their eyes on the fax machine for official documentation with photo ID and fingerprint any time a Merchant enters the sector.
* tweak: Mining access is required to claim money from the smelter (but not operate it)

Caboozles commented:
So with this, trading opens the way for more goodies, meaning even people who want goodies for themselves have a reason to trade?

Fund it!

kilozombie commented on code/game/jobs/job/whitelisted.dm:
```diff
@@ -65,9 +76,35 @@
 	to_chat(M, "<B>You are a [job_title].</B>")
 
-	to_chat(M, "<b>You've finally got your equipment together, such as it is. Now it's time for action and adventure! In the rush of excitement, you've forgotten where you were going to go. If only you had any friends that could remind you...</b>")
+	to_chat(M, "<b>You should do your best to sell what you can to fund new product sales. Ultimately, the mark of a good trader is profit -- but public relations are an important component of thatend goal.</b>")
```
"thatend"

kilozombie commented:
Massive fan of anything which moves Trader in a better direction and this looks great!
Only thing is, maybe this is an opportunity to really flesh out the trader vend? A more robust appearance, UI, location?, and some kind of periodically regenerating stock would help validate this "sell some stuff, go back and buy some more stuff, come back" cycle. I like changing it to monetary values, because this does what trader coin-based systems failed to do: give very cheap items a purpose and actual opportunity to obtain.

Things like materials with which to enhance the shuttle would be a great way to blow $10 or $50 at a time. Imagine if the trader vend had a column dedicated to Space Ikea, in which you can buy lots of furniture and occasionally some machinery?

I think with the changes you've laid out, as well as a host of new trader items that are niche/cheap as well as a few more that are great/expensive, Trader might finally become the job it was supposed to be.

Kurfursten commented:
MartianMadman has talked about a possible trader "cargo shuttle" or even a REVERSE shuttle where you fly your tradeship to a deaddrop/pickup spot in space. I think those are really cool ideas. I also have 11 new trader items in the works (nearly done) and more on the way after.

kilozombie commented:
Sounds wonderful. The idea of a trader cargo shuttle has been floating around since 2016, so that's exciting as hell.

otatoh commented:
Seconding Kilo's idea of adding furniture/customization for the shuttle. A way to replace the shuttle tiling short of building/deconstructing walls would be nice as well, so you can wrench machinery down on the shuttle (for traveling chef/cafe gimmicks and such). 

Ping me on d*scord if you want snowflake furniture sprites.

Durgen commented:
Looks good, but one thing I worry about people using meta knowledge that Merchants are non-antag although I guess that’s the point.

kilozombie commented:
Merchants should probably have Captain-equivalent antag chance so that you can't metagame. Maybe even less likely, but still possible, so there's always a sliver of doubt. (As in real life.)

Kurfursten commented:
OP updated with a new version of the notification fax. Improved formatting.

Reminder that the starting cash and prices for each thing in the vendor are all up for discussion.

chemist's pallet $200
wonderful wardrobe $350
mecha graveyard $100
bluespace beaker $100
bluespace crystals $150
borer egg $100
clown shoes $50
sea devil $50
matter synth $350

My general logic in this has been
* Things that need more work to be useful (e.g.: mecha graveyard, borer egg) at 100
* Things with a very narrow target audience ranging from 0 to 1 potential buyers at 50
* Things with a LOT of content at 350 meaning you either need to make some sales first or pool money with other traders

Kammerjunk commented:
Does this suggest a change in the station-side treatment of "Merchant" traders vs. "Trader" traders

Kurfursten commented:
That would not be my place to decide. Matters of that kind I think for good reason belong with the headmin. Since we lack one, I do not foresee any change in the immediate future.

bathosbathtime commented:
I like these changes.  I will miss vagabonds because I liked how the word sounds but that is a very small price to pay for a large step in the right direction for traders.

Dacendeth commented:
I'd like to see a space explorer themed bird title if you're removing my spacehobo title, but besides that this looks amazing.

kilozombie commented:
A fun third flavor specifically for space exploration might be Exile, arrest/kill on sight if steps foot on station (without very good reason) but given some kind of encouragement to go explore the stars instead?

AI-Is-Rogue commented:
Don't traders usually go vault hunting for things to sell? Being arrested or killed on sight would make it tough to peddle exotic goods from outer space.

Kurfursten commented:
I think ai-is-rogue is right. It's counter productive in that case.

Exxion commented:
Also, unless you disallowed them from defending themselves like with catbeasts, it would be a rules nightmare

Dacendeth commented:
Why is it never start as a antag instead of reduced like most other implanted jobs? Is Nanotrasen seriously putting more effort into making sure random birds are more trustworthy than their captains?

Durgen commented:
@kilozombie 
Smuggler would be a good name. Maybe they could be valid to sec only.

Kurfursten commented:
I can make the role "protected" but it's somewhat complicated by a lack of secret repo access to see exactly how that's implemented. The reason being that trader and merchant are not separate jobs from one another. It's easy to remove people with the alt title from antag eligibility altogether but I can talk to some collabs about secret repo access

strawberrytan commented:
How about the payoff from mining winking this a bit? It seems like a few minutes of dig will enable you to buy the entire vending machine and most(all?) maps feature ore processing for traders

Exxion commented:
@Kurfursten The secret repo isn't involved at all
Just search `protected_jobs`

Kammerjunk commented:
I was thinking that they'd just do pretty much exactly that (breaking in isn't a problem - no different than how traders behaved when the role first appeared), but the entire point is that they deal with cash so I didn't bring it up

Kurfursten commented:
@Exxion I don't think that's right because the only place protected_jobs is used is to add those roles to restricted based on a config setting, nothing to do with likelihood of selection at all. Unless that config value itself changes? I need to see where protected_jobs affects antag priority which must be on the secret repo if it does at all.

@Kammerjunk To be clear I don't object to making the mining computer ID locked at least as a stopgap as part of this PR until such a day as mining gets reworked. No reason to highly incentivize smash n grabbing.

strawberrytan commented:
What if it was a sort of soft ID lock in that trader IDs can't cash out, they need to go get a proper NT card to work that part

That way a vox could dig and so could a scientist without being boned

Exxion commented:
@Kurfursten There is no code on the secret repo dealing with antag selection, as far as I can see at least.

iFlashYou commented:
Oh nice, sec birds but with the power of traders

Kurfursten commented:
Here's what's new
* Fixed typo (thanks kilozombie)
* More clear if port is open or closed when looking at comms console (and in color!)
* Changed guaranteed nonantag significantly. Merchants will still **never** be starting cultists, vampires, or double agents (as sec/cap). However, they can now be traitors and changelings. It would seem "protected" jobs have the same chance of rolling these as anyone else and it was just a coincidence that not many heads/sec officers seem to roll these. This makes sense when you think about confirmation bias + small sample size of sec+cap + small sample size of antags + relatively few rounds are exactly traitor (not double agent) or ling. *When this gets merged I will put up a PR that people can vote on separately that would reduce the chances of a protected job/Merchant getting an antag roll.*
* Mining smelter computer can be operated by anyone but inserting an ID requires mining access. **Technical note**: This means that someone with a mining ID can insert a science ID, for example, as long as they're wearing the ID with appropriate access. Naturally this does not interfere with inserting an ID in your active hand either, even if you appear as "unknown" as long as that ID has mining access.
* Fixed a bug where only non-merchants would get the radio

@Exxion Thanks for letting me know regarding that, it makes sense the more I think about it. This PR is good to go.

Intigracy commented:
fyi server's current config

(E) (C) (M) (S) protect_roles_from_antagonist = 1 

anything on the protected lists doesn't get antag

Kurfursten commented:
Confusing if true because I have definitely seen sec traitors and they are on the protected traitor list

Exxion commented:
It is true. The code involved is also painfully awful, so if sec can indeed be traitors, it's probably a bug
Wonderful

Kurfursten commented:
Well, like I said, I can ungunk protected roles in a separate PR after this. It just doesn't make sense to be even less atomic when it seems like people could have very different thoughts about protected roles being antagonists.

Exxion commented:
Honestly I wouldn't bother with it this close to antag datums being finished

I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE commented:
Tradercult is back in fashion now that gult3 made loyalty implants useless, bravo

Kurfursten commented:
Literally not my fault, I have publicly petitioned Deity multiple times both in the Discord and on Github to return old loyalty behavior.

I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE commented:
Wasn't trying to make it sound like it was your fault, if it came across like that
